### PR TITLE
fix(FEC-9523): add attach detach implemention

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

for reloading media add attach detach implemention

Solve FEC-9523.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Comment by yairans** [comment]
@Yuvalke 
1. please add tests
2. remove the default implementation from [`BaseMediaSourceAdapter`](https://github.com/kaltura/playkit-js/blob/a45a556001598e5ef5e8c9dc2ed63bd78fe874e7/src/engines/html5/media-source/base-media-source-adapter.js#L180) and throw error like [this](https://github.com/kaltura/playkit-js/blob/a45a556001598e5ef5e8c9dc2ed63bd78fe874e7/src/engines/html5/media-source/base-media-source-adapter.js#L207)

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 1]
```diff
@@ -267,8 +281,9 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     if (!this._loadPromise) {
       this._loadPromise = new Promise((resolve, reject) => {
         this._lastTimeUpdate = startTime || 0;
+        const playbackStartTime = this._startTimeAttach || startTime || 0;
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: use `this. _lastTimeDetach`. `this. _startTimeAttach` is unnecessary.

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 2]
```diff
@@ -340,13 +355,23 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
    * @public
    * @returns {void}
    */
-  attachMediaSource(): void {}
+  attachMediaSource(): void {
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: you can leave it empty

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 3]
```diff
@@ -390,6 +415,7 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     } else {
       this._eventManager.listenOnce(this._videoElement, Html5EventType.CAN_PLAY, parseTracksAndResolve.bind(this));
     }
+    this._startTimeAttach = null;
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]:
```suggestion
    this. _lastTimeDetach = null;
```
actuaally i would reset it to `0` and type it as `number`

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 4]
```diff
@@ -117,6 +117,20 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
    */
   _mediaErrorRecoveryAttempts: number = 0;
 
+  /**
+   * The last time detach occurred
+   * @type {number}
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]:
```suggestion
   * @type {?number}
```
but see my comment below, that the default could be `0` and type is `number`

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 5]
```diff
@@ -340,13 +355,23 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
...
+    this._lastTimeDetach = null;
+  }
   /**
    * detach media - will remove the media source from handling the video
    * @public
    * @returns {void}
    */
-  detachMediaSource(): void {}
+  detachMediaSource(): void {
+    this._lastTimeDetach = this.currentTime;
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: clean on destroy

File: src/engines/html5/media-source/adapters/native-adapter.js [hunk 6]
```diff
@@ -340,13 +355,23 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
    * @public
    * @returns {void}
    */
-  attachMediaSource(): void {}
+  attachMediaSource(): void {
+    this._startTimeAttach = this._lastTimeDetach;
```
Comment (yairans) [inline src/engines/html5/media-source/adapters/native-adapter.js]: clean on destroy

File: src/engines/html5/media-source/adapters/native-adapter.js (hunk 1 above)
Comment (Yuvalke) [reply src/engines/html5/media-source/adapters/native-adapter.js]: Should keep the `this. _startTimeAttach` first cause if someone enter it as config it'll return always to this position

**Pull request merged**